src/LiveComponent/src/Util/QueryStringPropsExtractor.php

```diff
@@ -0,0 +1,87 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\UX\LiveComponent\Util;
+
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\UX\LiveComponent\Exception\HydrationException;
+use Symfony\UX\LiveComponent\LiveComponentHydrator;
+use Symfony\UX\LiveComponent\Metadata\LiveComponentMetadata;
+use Symfony\UX\LiveComponent\Metadata\LivePropMetadata;
+
+/**
+ * @author Nicolas Rigaud <squrious@protonmail.com>
+ *
+ * @experimental
+ *
+ * @internal
+ */
+final class QueryStringPropsExtractor
+{
+    public function __construct(private readonly LiveComponentHydrator $hydrator)
+    {
+    }
+
+    /**
+     * Extracts relevant query parameters from the current URL and hydrates them.
+     */
+    public function extract(Request $request, LiveComponentMetadata $metadata, object $component): array
+    {
+        $query = $request->query->all();
+
+        if (empty($query)) {
+            return [];
+        }
+        $data = [];
+
+        foreach ($metadata->getAllLivePropsMetadata() as $livePropMetadata) {
+            if ($queryStringMapping = $livePropMetadata->getQueryStringMapping()) {
+                if (null !== ($value = $query[$queryStringMapping['name']] ?? null)) {
+                    if (\is_array($value) && $this->isNumericIndexedArray($value)) {
+                        // Sort numeric array
+                        ksort($value);
+                    } elseif ('' === $value && null !== $livePropMetadata->getType() && (!$livePropMetadata->isBuiltIn() || 'array' === $livePropMetadata->getType())) {
+                        // Cast empty string to empty array for objects and arrays
+                        $value = [];
+                    }
+
+                    try {
+                        $hydratedValue = $this->hydrator->hydrateValue($value, $livePropMetadata, $component);
+
+                        if ($this->isValueTypeConsistent($hydratedValue, $livePropMetadata)) {
+                            // Only set data if hydrated value type is consistent with prop metadata type
+                            $data[$livePropMetadata->getName()] = $hydratedValue;
+                        }
```

Comment:
Would that not silent problems one could want to know about ?

Reply:
We discussed this point [here](https://github.com/symfony/ux/pull/1230#discussion_r1395754272). @weaverryan suggested that inconsistencies in data are not relevant to be propagated from there, because they come from a user input. Do you have other use cases in mind?

Reply:
This could be any attempt that should otherwise raise a security alert 

If someone add `&year=DROP TABLE;`  i think i'd like to know it :)

I'm particulary attentive there because the LiveComponent actions will soon be POST-only, protected by CSRF, etc.. 

But the moment we allow anyone to have a direct access to the data via a queryString can share this URL to make others the last "actor", that requires for me an extra layer of monitoring / checks.

Reply:
No but the validation is triggered if Asserts are defined and the trait is used :)